src/features/profile/pages/NotificationSettings/helpers/hasUserChangedParameters.ts

```diff
@@ -0,0 +1,21 @@
+import { UserProfileResponse } from 'api/gen'
+import { NotificationsSettingsState } from 'features/profile/pages/NotificationSettings/NotificationsSettings'
+import { SubscriptionTheme } from 'features/subscription/types'
+
+export const hasUserChangedParameters = (
+  user: UserProfileResponse,
+  state: NotificationsSettingsState
+): boolean => {
+  const hasChangeSubscriptions = !(
+    user?.subscriptions.subscribedThemes?.length === state.themePreferences.length &&
+    state.themePreferences.every((theme: SubscriptionTheme) =>
+      user?.subscriptions.subscribedThemes?.includes(theme)
+    )
+  )
+
+  return (
+    user?.subscriptions.marketingEmail !== !!state.allowEmails ||
+    user?.subscriptions.marketingPush !== !!state.allowPush ||
```

Comment:
Je ne pense pas qu'on ait besoin des doubles négations :
```suggestion
    user?.subscriptions.marketingEmail !== state.allowEmails ||
    user?.subscriptions.marketingPush !== state.allowPush ||
```

Reply:
ça pose des problèmes de typing lorsque l'utilisateur est pas encore présent les state allowEmail et allowPush sont undefined, et j'ai envie de considérer que c'est false sinon le bouton est actif pendant ce temps là